src/rpc/mempool.cpp

```diff
@@ -124,6 +124,7 @@ static RPCHelpMan testmempoolaccept()
                     {RPCResult::Type::OBJ, "fees", /*optional=*/true, "Transaction fees (only present if 'allowed' is true)",
                     {
                         {RPCResult::Type::STR_AMOUNT, "base", "transaction fee in " + CURRENCY_UNIT},
+                        {RPCResult::Type::STR_AMOUNT, "effective-feerate", /*optional=*/true, "if something other than base feerate was used, the effective feerate in " + CURRENCY_UNIT + " per KvB. For example, the package feerate and/or feerate with modified fees from prioritisetransaction."},
```

Comment:
> Have you considered always including this field?

We don't always know it. We could have quit validation before looking up inputs, e.g. because `CheckTransaction()` failed

Reply:
But I think we always know it when `allowed=true`? For `fees`, the docs already state that "(only present if 'allowed' is true)".  So I think, consistent with `base` (which isn't optional), we can always (when `allowed=true`) return `effective-feerate`. This makes for a more consistent API imo.

I suppose this also comes down to personal preference, so I'm okay leaving it as is too if you/others prefer.

Reply:
Changed to non-optional now